src/functions/module_lookup.ts

```diff
@@ -170,7 +170,12 @@ export const LOOKUP: AddFunctionDescription = {
       rangeLength,
       getElement
     );
-    assertAvailable(searchArray[0][index], searchKey);
+
+    if (resultRange === undefined && !verticalSearch && index !== -1) {
```

Comment:
Why `resultRange === undefined` ?
No matter if `resultRange` is defined or not, the search key must be found in the search array.
And when removing it all tests pass.

All in all, I think only `if (!verticalSearch) {` remains. Then both cases can be switched to avoid the negation in the if predicate. Negations are harder to read ;)